src/DeterministicNoiseSimulator.cpp

```diff
@@ -4,680 +4,77 @@
     }
-    dd::Edge<dd::dNode> e = {};
-    if (std::count(used_qubits.begin(), used_qubits.end(), originalCopy.p->v)) {
-        for (auto& new_edge: new_edges) {
-            if (new_edge.w.approximatelyZero()) {
-                new_edge.w = dd::Complex::zero;
-            } else {
-                new_edge.w = dd->cn.getCached(dd::CTEntry::val(new_edge.w.r), dd::CTEntry::val(new_edge.w.i));
-            }
-        }
 
-        for (auto const& type: gateNoiseTypes) {
-            switch (type) {
-                case 'A':
-                    ApplyAmplitudeDampingToNode(new_edges, (used_qubits.size() == 1) ? ampDampingProbSingleQubit : ampDampingProbMultiQubit);
-                    break;
-                case 'P':
-                    ApplyPhaseFlipToNode(new_edges, (used_qubits.size() == 1) ? noiseProbSingleQubit : noiseProbMultiQubit);
-                    break;
-                case 'D':
-                    ApplyDepolarisationToNode(new_edges, (used_qubits.size() == 1) ? noiseProbSingleQubit : noiseProbMultiQubit);
-                    break;
-                default:
-                    throw std::runtime_error(std::string("Unknown gate noise type '") + type + "'");
-            }
-        }
-
-        for (auto& new_edge: new_edges) {
-            if (new_edge.w.approximatelyZero()) {
-                if (new_edge.w != dd::Complex::zero) {
-                    dd->cn.returnToCache(new_edge.w);
-                    new_edge.w = dd::Complex::zero;
-                }
-            } else {
-                dd::Complex c = dd->cn.lookup(new_edge.w);
-                dd->cn.returnToCache(new_edge.w);
-                new_edge.w = c;
-            }
-        }
-    }
-
-    e = dd->makeDDNode(originalCopy.p->v, new_edges, false, firstPathEdge);
-
-    // Adding the noise operation to the cache, note that e.w is from the complex number table
-    //    dd->densityNoise.insert(originalCopy, e, used_qubits);
-
-    // Multiplying the old edge weight with the new one and looking up in the complex numbers table
-    if (!e.w.approximatelyZero()) {
-        if (e.w.approximatelyOne()) {
-            e.w = originalEdge.w;
-        } else {
-            auto tmpComplexValue = dd->cn.getCached();
-            CN::mul(tmpComplexValue, e.w, originalEdge.w);
-            e.w = dd->cn.lookup(tmpComplexValue);
-            dd->cn.returnToCache(tmpComplexValue);
-        }
-    }
-    return e;
-}
-
-void DeterministicNoiseSimulator::ApplyPhaseFlipToNode(std::array<dEdge, 4>& e, double probability) {
-    dd::Complex complex_prob = dd->cn.getCached();
-
-    //e[0] = e[0]
-
-    //e[1] = (1-2p)*e[1]
-    if (!e[1].w.approximatelyZero()) {
-        complex_prob.r->value = 1 - 2 * probability;
-        complex_prob.i->value = 0;
-        CN::mul(e[1].w, complex_prob, e[1].w);
+    // Create the final map containing the measurement results and the corresponding shots
+    std::map<std::string, std::size_t> resultShotsMap;
+    for (auto& res: results) {
```

Comment:
as above: use `const` and structured bindings to make the intent clear

Reply:
done